tests/test_bad.py

```diff
@@ -0,0 +1,136 @@
+        ----------
+        config_name: `str`
+            name of the OODS configuration file
+        fits_name: `str`
+            name of the test FITS file
+
+        Returns
+        -------
+        config: `dict`
+            An OODS configuration to use for testing
+        """
+
+        # create a path to the configuration file
+
+        testdir = os.path.abspath(os.path.dirname(__file__))
+        configFile = os.path.join(testdir, "etc", config_name)
+
+        # path to the FITS file to ingest
+
+        fitsFile = os.path.join(testdir, "data", fits_name)
+
+        # load the YAML configuration
+
+        with open(configFile, "r") as f:
+            config = yaml.safe_load(f)
+
+        # extract parts of the ingester configuration
+        # and alter the image staging directory to point
+        # at the temporary directories created for his test
+
+        ingesterConfig = config["ingester"]
+        self.imageDir = tempfile.mkdtemp()
+        ingesterConfig["imageStagingDirectory"] = self.imageDir
+
+        self.badDir = tempfile.mkdtemp()
+        butlerConfig = ingesterConfig["butlers"][0]["butler"]
+        butlerConfig["badFileDirectory"] = self.badDir
+        self.stagingDir = tempfile.mkdtemp()
+        butlerConfig["stagingDirectory"] = self.stagingDir
+
+        self.repoDir = tempfile.mkdtemp()
+        butlerConfig["repoDirectory"] = self.repoDir
+
+        # copy the FITS file to it's test location
+
+        self.subDir = tempfile.mkdtemp(dir=self.imageDir)
+        self.destFile = os.path.join(self.subDir, fits_name)
+        shutil.copyfile(fitsFile, self.destFile)
+
+        return config
+
+    def tearDown(self):
+        """Remove directories created by createConfig"""
+        shutil.rmtree(self.imageDir, ignore_errors=True)
+        shutil.rmtree(self.badDir, ignore_errors=True)
+        shutil.rmtree(self.stagingDir, ignore_errors=True)
+        shutil.rmtree(self.repoDir, ignore_errors=True)
+        shutil.rmtree(self.subDir, ignore_errors=True)
+
+    async def testBadIngest(self):
+        """test ingesting a bad file"""
+        fits_name = "AT_O_20221122_000951_R00_S00.fits.fz"
+        config = self.createConfig("ingest_auxtel_gen3.yaml", fits_name)
+
+        # setup directory to scan for files in the image staging directory
+        ingesterConfig = config["ingester"]
+        image_staging_dir = ingesterConfig["imageStagingDirectory"]
+        scanner = DirectoryScanner([image_staging_dir])
+        files = scanner.getAllFiles()
+        self.assertEqual(len(files), 1)
+
+        ingester = FileIngester(config["ingester"])
+
+        staged_files = ingester.stageFiles([self.destFile])
+        await ingester.ingest(staged_files)
+
+        files = scanner.getAllFiles()
+        self.assertEqual(len(files), 0)
+
+        name = Utils.strip_prefix(self.destFile, image_staging_dir)
+        bad_path = os.path.join(self.badDir, name)
+        self.assertTrue(os.path.exists(bad_path))
```

Comment:
You don't actually check that error messages are emitted, though?  That's probably OK (it's good enough that it doesn't crash), but might be a future enhancement.

Reply:
The main purpose was that it didn't crash as you said, and that the file showed up in the "bad files" directory.  I'll get the code refactored later to test the error message.